### PR TITLE
Adds connect with idtoken

### PR DESCRIPTION
**Pull request by yashovardhan**
This adds the function for logging in directly using an idtoken, extending the getTorusKey functionalities.

### CODE REVIEW

File: src/interfaces.ts [hunk 1]
```diff
@@ -29,32 +37,76 @@ export interface AggregateVerifierLoginParams extends BaseLoginParams {
...
+   */
+  additionalParams?: ExtraParams;
+}
+
 // TODO extend interface and type documentation wherever necessary. @Yash
 export interface ICoreKit {
-  /** The tKey instance, if initialized. */
+  /** The tKey SDK instance
+   *  TKey is the core module on which this wrapper SDK sits for easy integration.
+   **/
```
Comment (matthiasgeihs) [inline src/interfaces.ts:76]: This needs to be formatted according to standard.
```suggestion
  /**
   * The tKey instance, if initialized.
   * TKey is the core module on which this wrapper SDK sits for easy integration.
   **/
```

File: src/interfaces.ts [hunk 2]
```diff
@@ -29,32 +37,76 @@ export interface AggregateVerifierLoginParams extends BaseLoginParams {
...
   // codes and document which are thrown. in particular here, error is thrown if
   // not factor key is given (either as function parameter or through local
   // storage)
-  // TODO login with id token @Yash
   /**
-   * Login to tKey and initialize all relevant components.
-   * @param loginParams - TKey login parameters.
+   * Login into the SDK in an implicit flow and initialize all relevant components.
+   * @param loginParams - Parameters for Implicit Login.
+   * @param factorKey - A BN used for encrypting your Device/ Recovery TSS Key Share. Optional for new users, mandatory for existing users, if not provided we will try to fetch it from local storage. Read more about it here: https://web3auth.io/docs/infrastructure/mpc-architecture#factor-keys
```
Comment (matthiasgeihs) [inline src/interfaces.ts:86]: I suggest we don't include links to docs here, as these can easily break in the future. The developer can always find the docs via website.

File: src/interfaces.ts [hunk 3]
```diff
@@ -29,32 +37,76 @@ export interface AggregateVerifierLoginParams extends BaseLoginParams {
...
    * @returns A Web3 provider if we are not in redirect mode.
    */
   login(loginParams: LoginParams, factorKey?: BN): Promise<SafeEventEmitterProvider | null>;
 
+  /**
+   * Login into the SDK using ID Token based login and initialize all relevant components.
+   * @param idTokenLoginParams - Parameters with ID Token based Login.
+   * @param factorKey - A BN used for encrypting your Device/ Recovery TSS Key Share. Optional for new users, mandatory for existing users, if not provided we will try to fetch it from local storage. Read more about it here: https://web3auth.io/docs/infrastructure/mpc-architecture#factor-keys
```
Comment (matthiasgeihs) [inline src/interfaces.ts:94]: Remove link, see comment above.

File: src/interfaces.ts [hunk 4]
```diff
@@ -29,32 +37,76 @@ export interface AggregateVerifierLoginParams extends BaseLoginParams {
...
+   * @returns A Web3 provider
+   */
+  loginWithIdToken(idTokenLoginParams: IdTokenLoginParams, factorKey?: BN): Promise<SafeEventEmitterProvider | null>;
+
   /**
    * Handle redirect result after login.
-   * @param factorKey - An optional factor key to use at initialization.
+   * @param factorKey - A BN used for encrypting your Device/ Recovery TSS Key Share. Optional for new users, mandatory for existing users, if not provided we will try to fetch it from local storage. Read more about it here: https://web3auth.io/docs/infrastructure/mpc-architecture#factor-keys
```
Comment (matthiasgeihs) [inline src/interfaces.ts:101]: Remove link, see comment above.

File: src/interfaces.ts [hunk 5]
```diff
@@ -29,32 +37,76 @@ export interface AggregateVerifierLoginParams extends BaseLoginParams {
...
    * @returns A Web3 provider.
    */
   handleRedirectResult(factorKey?: BN): Promise<SafeEventEmitterProvider>;
 
   /**
    * Indicates whether there is an existing session that can be resumed.
+   * @returns A boolean `true` indicating that there is an existing session that can be resumed.
```
Comment (matthiasgeihs) [inline src/interfaces.ts:108]: Ideally comments should wrap at 80 chars.
You can use VSCode extension `rewrap` for this.

File: src/interfaces.ts [hunk 6]
```diff
@@ -65,16 +117,16 @@ export interface ICoreKit {
...
   /**
    * Creates a new factor for authentication.
-   * @param factorKey - The factor key.
-   * @param shareType - The share type.
-   * @param shareDescription - The share description.
-   * @param additionalMetadata - Additional metadata.
+   * @param factorKey - A BN used for encrypting your Device/ Recovery TSS Key Share. You can generate it using `generateFactorKey()` function or use an existing one.
+   * @param shareType - Setting the Type of Share - Device or Recovery. Default is DEVICE.
+   * @param shareDescription - Setting the Description of Share - Security Questions, Device Share, Seed Phrase, Password Share, Social Share, Other. Default is Other.
+   * @param additionalMetadata - Additional metadata information you want to be stored alongside this factor for easy identification.
```
Comment (matthiasgeihs) [inline src/interfaces.ts:129]: Nice!

File: src/interfaces.ts [hunk 7]
```diff
@@ -94,17 +146,19 @@ export interface ICoreKit {
 
   /**
    * Generates a new factor key.
+   * A factor key is a BN used for encrypting your Device/ Recovery TSS Key Shares.
+   * Read more about it here: https://web3auth.io/docs/infrastructure/mpc-architecture#factor-keys
```
Comment (matthiasgeihs) [inline src/interfaces.ts:150]: Remove link

File: src/mpcCoreKit.ts [hunk 8]
```diff
@@ -180,6 +184,55 @@ export class Web3AuthMPCCoreKit implements ICoreKit {
     }
   }
 
+  public async loginWithIdToken(idTokenLoginParams: IdTokenLoginParams, factorKey: BN | undefined = undefined) {
```
[the remark this answers is not in the document]
Comment (yashovardhan) [reply src/mpcCoreKit.ts:187]: I understand the point, however I think just by wrapping within the main login function, we might complicate the SDK a bit. I additionally would like @himanshuchawla009 and @ieow to comment on what they think about about the structure as well.

[the remark this answers is not in the document]
Comment (ieow) [reply src/mpcCoreKit.ts:184]: I would suggest to keep the idToken login's name as `login` and change current `login` to `loginOAuth`

[the remark this answers is not in the document]
Comment (yashovardhan) [reply src/mpcCoreKit.ts:184]: I agree

**Review by ieow — APPROVED** [review]
lgtm

**Comment by ieow** [comment]
I need to merge this and rebase my pr on this

**Pull request merged**